enterprise/server/scheduling/priority_task_scheduler/priority_task_scheduler.go

```diff
@@ -594,6 +720,56 @@ func (q *PriorityTaskScheduler) trimQueue() bool {
 	return true
 }
 
+// queuePosition points to a specific position in the global task queue ordering
+// (across all groups), allowing immediate access to the element at that
+// position.
+//
+// If the queue is modified, the queuePosition is no longer valid.
+type queuePosition struct {
+	// Which of the per-group queues the task is in.
+	GroupQueue *list.Element
+	// The index of the task within the GroupQueue.
+	Index int
+}
+
+// getNextSchedulableTask returns the next task that can be scheduled, and a
+// pointer to the task in the queue.
+func (q *PriorityTaskScheduler) getNextSchedulableTask() (*scpb.EnqueueTaskReservationRequest, *queuePosition) {
+	// Don't use the experimental custom resource scheduling logic if there are
+	// no custom resources configured.
+	if len(q.customResourcesCapacity) == 0 {
+		task := q.q.Peek()
+		if task == nil {
+			return nil, nil
+		}
+		if canFit, _ := q.canFitTask(task); canFit {
+			return task, q.q.headRef()
+		}
+		return nil, nil
```

Comment:
does the old logic not work here?

```
	nextTask := q.q.Peek()
	if nextTask == nil || !q.canFitTask(nextTask) {
		return nil, nil
	}
        return nextTask, q.q.headRef()
```

Reply:
it doesn't quite work since canFit returns 2 bools now. Negated the `canFit` check though to make it match the old implementation more closely